### PR TITLE
[zero] polish ZeroInitContext

### PR DESCRIPTION
**Pull request by ver217**
1. Remove `convert_to_fp16` argument. We don't support fp32 training, parameters must be converted to fp16.
2. Remove `shard_grad` argument. We only support ZeRO-2 and ZeRO-3. Gradients are always sharded.
3. Set the default dtype of `model_numel_tensor` to `torch.long`, in case of overflow.
4. Polish corresponding unit tests.

### CODE REVIEW

File: colossalai/zero/init_ctx/init_context.py [hunk 1]
```diff
@@ -4,12 +4,12 @@
 import torch
 from colossalai.context.parallel_mode import ParallelMode
 from colossalai.core import global_context as gpc
+from colossalai.logging import disable_existing_loggers, get_dist_logger
```
Comment (FrankLeeeee) [inline colossalai/zero/init_ctx/init_context.py:7]: `disable_existing_loggers` is not used in this file and can be removed.

**Comment by FrankLeeeee** [comment]
LGTM.

File: colossalai/zero/init_ctx/init_context.py (hunk 1 above)
Comment (ver217) [reply colossalai/zero/init_ctx/init_context.py]: DONE

**Review by feifeibear — COMMENTED** [review]
Did you update examples correspondingly?

**Comment by ver217** [comment]
> Did you update examples correspondingly?

I will create a PR ASAP.
